examples/magento-graphcms/components/ProductListItems/ProductScroller.tsx

```diff
@@ -0,0 +1,118 @@
+import {
+  AddProductsToCartContext,
+  AddProductsToCartForm,
+  AddProductsToCartFormProps,
+  ProductListQuery,
+} from '@graphcommerce/magento-product'
+import {
+  ItemScroller,
+  ItemScrollerProps,
+  RenderType,
+  nonNullable,
+  responsiveVal,
+} from '@graphcommerce/next-ui'
+import {
+  Box,
+  Container,
+  ContainerProps,
+  Skeleton,
+  SxProps,
+  Theme,
+  Typography,
+  TypographyProps,
+  useTheme,
+} from '@mui/material'
+import { useContext } from 'react'
+import { productListRenderer } from './productListRenderer'
+import React from 'react'
+
+export function ProductScrollerItemSkeleton({ imageOnly = false }: { imageOnly?: boolean }) {
+  return (
+    <Box>
+      <Skeleton sx={{ width: '100%', aspectRatio: '1/1', transform: 'none' }} />
+
+      {!imageOnly && (
+        <Typography
+          variant='subtitle1'
+          sx={(theme) => ({
+            marginTop: theme.spacings.xs,
+            display: 'block',
+            color: 'text.primary',
+            overflowWrap: 'break-word',
+            wordBreak: 'break-all',
+            maxWidth: '100%',
+            gridArea: 'title',
+            fontWeight: 'fontWeightBold',
+          })}
+        >
+          &nbsp;
+        </Typography>
+      )}
+    </Box>
+  )
+}
+
+export function ProductScroller({
+  title = '',
+  items,
+  imageOnly = false,
+  skeletonItemCount = 0,
+  skeleton,
+  sx = [],
+  containerProps,
+  titleProps,
+  itemScrollerProps,
+  addProductsToCartFormProps,
+}: {
+  title?: string
+  items: Exclude<ProductListQuery['products'], null | undefined>['items']
+  imageOnly?: boolean
+  skeletonItemCount: number
+  skeleton?: React.ReactNode
+  sx?: SxProps<Theme>
+  containerProps?: ContainerProps
+  titleProps?: TypographyProps
+  addProductsToCartFormProps?: AddProductsToCartFormProps
+  itemScrollerProps?: ItemScrollerProps
+}) {
+  const theme = useTheme()
+
+  if (!!useContext(AddProductsToCartContext) && process.env.NODE_ENV !== 'production')
+    throw new Error("Can't use ProductScroller inside of AddProductsToCartForm")
```

Comment:
It is bad practice to throw during development an silence during production, also you are wrapping in AddProductsToCartForm below, so I don't think it is necessary that is wrapped inside one.

Reply:
Fixed by conditionally wrapping in AddProductsToCartForm based on AddProductsToCartContext availability